third_party/amd/backend/compiler.py

```
@@ -70,6 +70,12 @@ def __post_init__(self):
         # Ignore user-defined warp size for gfx9
         warp_size = 32 if 'gfx10' in self.arch or 'gfx11' in self.arch or 'gfx12' in self.arch else 64
         object.__setattr__(self, 'warp_size', warp_size)
+        # Error if max threads/block is exceeded
+        # Architecture-dependent, check arch in future if values other than 1024 become supported
+        max_threads = 1024
+        assert self.num_warps * warp_size <= max_threads, \
```

can we raise an exception instead? This would allow it to work with the auto-tuner

---

This should raise an AssertionError exception already; is there something specific that needs to be thrown to work with the auto-tuner? `raise ValueError(f"{self.num_warps} warps * {warp_size} warp size must not exceed the max threads per block limit ({max_threads})")` would probably also be appropriate here.

---

ideally should be a `triton.CompilationError`

---

the auto-tuner catches those:
```
except (OutOfResources, CompileTimeAssertionFailure, PTXASError)
```
however now I realize none of them are a good match here, so feel free to ignore and it can solved later if someone runs into that

---

maybe CompilationError would be more accurate but that's really a detail

---

`OutOfResources` might be a good one? We are using more threads than the CU hardware can host effectively.

---

Makes sense. Getting a recursive import error when trying to `from triton.compiler.errors import OutOfResources` though.

---

Okay this is fine for now then.